runtime/component/visibility.js

```diff
@@ -0,0 +1,59 @@
+var _ = require('@yoda/util')._
+var manifest = require('@yoda/manifest')
+var logger = require('logger')('visibility')
+
+var VisibilityHandler = {
+  voice: function () {
+    Object.defineProperty(this, 'keyAndVisibleAppId', {
+      enumerable: true,
+      get: () => {
+        return _.get(this.component.audioFocus.getCurrentFocus(), 'appId')
+      }
+    })
+    Object.defineProperty(this, 'visibilities', {
+      enumerable: true,
+      get: () => {
+        return [ this.component.audioFocus.transientRequest, this.component.audioFocus.lastingRequest ]
+          .filter(it => it != null)
+          .map(it => it.appId)
+      }
+    })
+
+    this.abandonKeyVisibility = () => {
+      return this.component.audioFocus.abandonCurrentFocus()
+    }
+    this.abandonAllVisibilities = () => {
+      return this.component.audioFocus.abandonAllFocuses()
+    }
+  }
+}
+
+class Visibility {
+  constructor (runtime) {
+    this.runtime = runtime
+    this.component = runtime.component
+
+    var availableVisibilities = Object.keys(VisibilityHandler)
+    var mainVisibility = manifest.get('capabilities.main-visibility', 'voice')
+    if (availableVisibilities.indexOf(mainVisibility) < 0) {
+      logger.error(`Unknown main-visibility manifest '${mainVisibility}'`)
+      return
+    }
+    var handler = VisibilityHandler[mainVisibility]
+    handler.apply(this)
+  }
+
+  getKeyAndVisibleAppId () {
+    return this.keyAndVisibleAppId
```

Comment:
It's better to declare the property `keyAndVisibleAppId` inside the class `Visibility`.